## Estadística e Inferencia II - 2do cuatrimestre 2025
### Escuela de Ciencia y Tecnología - Universidad Nacional de San Martín
#### Javier Spina - javierspina@gmail.com | jaspina@estudiantes.unsam.edu.ar

---

## Trabajo Práctico Integrador - Parte 1
# Análisis Bayesiano del Dataset Iris

Este notebook cubre las consignas especificadas en `consignas/TP_integrador_EeI_II__parte_1.pdf`

---

### Parte 1 - Análisis exploratorio del dataset IRIS

In [1]:
import os
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.datasets import load_iris

sns.set_theme('notebook')

In [2]:
iris = load_iris(as_frame=True).frame
iris['species'] = iris['target'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})
iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    int64  
 5   species            150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


In [3]:
iris.describe()

sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)      target  
count        150.000000  150.000000  
mean           1.199333    1.000000  
std            0.762238    0.819232  
min            0.100000    0.000000  
25%            0.300000    0.000000  
50%            1.300000    1.000000  
75%            1.800000    2.000000  
max            2.500000    2.000000

In [4]:
iris.species.value_counts()

species
setosa        50
versicolor    50
virginica     50
Name: count, dtype: int64

El dataset `iris` tiene 150 observaciones, 4 features y un target. Las 4 features son numéricas, son las medidas del sépalo y del pétalo, en ancho y longitud de ambas. El target es el nombre de la variedad, osea es categórica. En la siguiente tabla un breve resumen

<style>
td, th {
   border: none!important;
}
</style>

Nombre | Es target? | Tipo | Rango 
-- | -- | -- | -- 
`sepal length (cm)` | no | `float` | $\in [4.3, 7.9]$
`sepal width (cm)` | no | `float` | $\in [2.0, 4.4]$
`petal length (cm)` | no | `float` | $\in [1.0, 6.9]$
`petal width (cm)` | no | `float` | $\in [0.1, 2.5]$
`species` | si | `object` | $\in [\text{setosa}, \text{versicolor}, \text{virginica}]$

**Variables numéricas**

Feature | Rango | Media | Mediana | Desvío
-- | -- | -- | -- | --
`sepal length (cm)` | $\in [4.3, 7.9]$ | $5.84$ | $5.80$ | $0.83$
`sepal width (cm)` | $\in [2.0, 4.4]$ | $3.06$ | $3.00$ | $0.44$
`petal length (cm)` | $\in [1.0, 6.9]$ | $3.76$ | $4.35$ | $1.76$
`petal width (cm)` | $\in [0.1, 2.5]$ | $1.20$ | $1.30$ | $0.76$

**Variable categórica (target)**
Variable | Valor | Frecuencia
-- | -- | -- 
`species` | $\text{setosa}$ | $50$ 
`species` | $\text{versicolor}$ | $50$ 
`species` | $\text{virginica}$ | $50$ 